src/wallet/walletdb.cpp

```diff
@@ -141,6 +159,18 @@ bool CWalletDB::WriteZKey(const libzcash::SproutPaymentAddress& addr, const libz
     // pair is: tuple_key("zkey", paymentaddress) --> secretkey
     return Write(std::make_pair(std::string("zkey"), addr), key, false);
 }
+bool CWalletDB::WriteSaplingZKey(const libzcash::SaplingPaymentAddress &defaultAddr,
+                                const libzcash::SaplingIncomingViewingKey &ivk,
+                                const libzcash::SaplingExtendedSpendingKey &key,
+                                const CKeyMetadata &keyMeta )
+{
+    nWalletDBUpdated++;
+    
+    if (!Write(std::make_pair(std::string("sapzkeymeta"), defaultAddr ), keyMeta))
```

Comment:
This is incorrectly using `defaultAddr` in the key, while the read side correctly expects `ivk`.